daemon/volumes.go

```diff
@@ -59,31 +59,31 @@ func (m mounts) parts(i int) int {
 // 2. Select the volumes mounted from another containers. Overrides previously configured mount point destination.
 // 3. Select the bind mounts set by the client. Overrides previously configured mount point destinations.
 // 4. Cleanup old volumes that are about to be reassigned.
-func (daemon *Daemon) registerMountPoints(container *container.Container, hostConfig *containertypes.HostConfig) (retErr error) {
+func (daemon *Daemon) registerMountPoints(ctx context.Context, container *container.Container, hostConfig *containertypes.HostConfig) (retErr error) {
 	binds := map[string]bool{}
 	mountPoints := map[string]*volumemounts.MountPoint{}
 	parser := volumemounts.NewParser()
 
-	ctx := context.TODO()
 	defer func() {
 		// clean up the container mountpoints once return with error
 		if retErr != nil {
 			for _, m := range mountPoints {
 				if m.Volume == nil {
 					continue
 				}
-				daemon.volumes.Release(ctx, m.Volume.Name(), container.ID)
+				daemon.volumes.Release(context.TODO(), m.Volume.Name(), container.ID)
 			}
 		}
 	}()
 
-	dereferenceIfExists := func(destination string) {
+	dereferenceIfExists := func(ctx context.Context, destination string) error {
 		if v, ok := mountPoints[destination]; ok {
 			logrus.Debugf("Duplicate mount point '%s'", destination)
 			if v.Volume != nil {
-				daemon.volumes.Release(ctx, v.Volume.Name(), container.ID)
+				return daemon.volumes.Release(ctx, v.Volume.Name(), container.ID)
```

Comment:
I see we previously ignored the error here; might be good to check if there was a specific reason for that (and if there was, definitely warrants a comment); code was added in https://github.com/moby/moby/commit/9526e5c6aebfaf8c8d18cefa0cf44e92b7ad5f77, but we need to check if it was not ignore _on purpose_ (perhaps some race conditions, e.g.)

Reply:
`daemon.volumes.Dereference` in https://github.com/moby/moby/commit/9526e5c6aebfaf8c8d18cefa0cf44e92b7ad5f77 didn't have an error return value to ignore: https://github.com/moby/moby/blob/9526e5c6aebfaf8c8d18cefa0cf44e92b7ad5f77/volume/store/store.go#L580-L581
Then in https://github.com/moby/moby/pull/36688 `Dereference` was refactored to become `Release`, taking a context argument and returning an error. The only errors the implementation returns come from `ctx.Err()` so the previously-ignored error was guaranteed to be `nil` since the context argument was `context.TODO()`.
https://github.com/moby/moby/blob/7691d17a37990b01386716fce8d76e96b90b0a7f/volume/service/store.go#L831-L854